google-cloud-pubsub/lib/google/cloud/pubsub/batch_publisher.rb

```diff
@@ -39,11 +39,26 @@ class BatchPublisher
         # @private The messages to publish
         attr_reader :messages
 
+        ##
+        # @private Enables publisher compression
+        attr_reader :compress
+
+        ##
+        # @private The threshold bytes size for compression
+        attr_reader :compression_bytes_threshold
+
+        ##
+        # @private The total bytes size of messages data.
+        attr_reader :total_message_bytes
+
         ##
         # @private Create a new instance of the object.
-        def initialize data, attributes, ordering_key, extra_attrs
+        def initialize data, attributes, ordering_key, extra_attrs, compress: false, compression_bytes_threshold: 240
```

Comment:
It would be good for the default `compression_bytes_threshold` to be a constant somewhere, so the 240 literal value isn't repeated in multiple places (and risks getting out of sync if it ever gets changed)

Reply:
I will probably update to have a parent class which can hold the common things for BatchPublisher and AsyncPublisher.

Reply:
@dazuma Please review the changes with new `Publisher` class